l1-contracts/src/core/libraries/validator-selection/ValidatorSelectionLib.sol

```diff
@@ -178,8 +179,10 @@ library ValidatorSelectionLib {
     // We do -1, as the snapshots practically happen at the end of the block, e.g.,
     // a tx manipulating the set in at $t$ would be visible already at lookup $t$ if after that
     // transactions. But reading at $t-1$ would be the state at the end of $t-1$ which is the state
-    //  as we "start" time $t$.
-    uint32 ts = Timestamp.unwrap(_epoch.toTimestamp()).toUint32() - 1;
+    // as we "start" time $t$. We then shift that back by an entire L2 epoch to guarantee
+    // we are not hit by last-minute changes or L1 reorgs when syncing validators from our clients.
+    uint32 ts =
+      Timestamp.unwrap(_epoch.toTimestamp()).toUint32() - uint32(TimeLib.getEpochDuration()) - 1;
```

Comment:
Epoch duration is not number of seconds in the epoch 😬 (Bad name). So this one right now subtract 33 seconds 🤔